From 1449f1e4270101f68185517554bffe29ba815f14 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Fri, 5 Jul 2019 23:06:12 +0200
Subject: [PATCH 113/121] dt-bindings: irqchip: document the bindings for the
 Lantiq XWAY EIU

The Lantiq External Interrupt Unit (EIU) is a GPIO interrupt-controller.
It maps up to six GPIOs each to one parent interrupt.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 .../interrupt-controller/lantiq,xway-eiu.yaml | 69 +++++++++++++++++++
 1 file changed, 69 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/interrupt-controller/lantiq,xway-eiu.yaml

diff --git a/Documentation/devicetree/bindings/interrupt-controller/lantiq,xway-eiu.yaml b/Documentation/devicetree/bindings/interrupt-controller/lantiq,xway-eiu.yaml
new file mode 100644
index 000000000000..c32102100bdd
--- /dev/null
+++ b/Documentation/devicetree/bindings/interrupt-controller/lantiq,xway-eiu.yaml
@@ -0,0 +1,69 @@
+# SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: "http://devicetree.org/schemas/interrupt-controller/lantiq,xway-eiu.yaml#"
+$schema: "http://devicetree.org/meta-schemas/core.yaml#"
+
+title: Lantiq XWAY External Interrupt Unit
+
+description: |
+  This interrupt controller is found in the Lantiq XWAY SoCs.
+  It is the GPIO interrupt controller, also called "EIU" (which
+  stands for External Interrupt Unit).
+
+allOf:
+  - $ref: /schemas/interrupt-controller.yaml#
+
+maintainers:
+  - Martin Blumenstingl <martin.blumenstingl@googlemail.com>
+
+properties:
+  compatible:
+    items:
+      - enum:
+          - lantiq,ase-eiu
+          - lantiq,danube-eiu
+          - lantiq,xrx100-eiu
+          - lantiq,xrx200-eiu
+          - lantiq,xrx300-eiu
+      - const: lantiq,xway-eiu
+
+  reg:
+    maxItems: 1
+
+  lantiq,eiu-irqs:
+    $ref: /schemas/types.yaml#/definitions/uint32-array
+    minItems: 1
+    maxItems: 6
+    description: One ICU interrupt specifier for each EIU
+
+  interrupt-controller: true
+
+  '#interrupt-cells':
+    const: 2
+
+  '#address-cells':
+    const: 2
+
+required:
+  - compatible
+  - reg
+  - lantiq,eiu-irqs
+  - interrupt-controller
+  - '#interrupt-cells'
+  - '#address-cells'
+
+additionalProperties: false
+
+examples:
+  - |
+    interrupt-controller@101000 {
+      compatible = "lantiq,xrx200-eiu", "lantiq,xway-eiu";
+      reg = <0x101000 0x10>;
+      interrupt-parent = <&icu0>;
+      lantiq,eiu-irqs = <158>, <127>, <58>, <32>, <33>, <34>;
+      interrupt-controller;
+      #interrupt-cells = <2>;
+      #address-cells = <2>;
+    };
+...
-- 
2.47.2

